### PR TITLE
 adapter: allow suppressing warnings for missing "current" objects

### PR DESCRIPTION
**Pull request by benesch**
Materialize will generally attempt to warn the user if their session's
"current" database, schema, or cluster does not exist via the pgwire
notices.

However, not all applications want these notices. In particular, they're
quite spammy with dbt, which often establishes dozens of connections
with a known invalid cluster, which it ignores because it always runs
DDL commands with the cluster explicitly specified.

Fix https://github.com/MaterializeInc/materialize/issues/28857.

### Motivation

  * This PR fixes a recognized bug.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

**Review by bobbyiliev — APPROVED** [review]
Nice!

File: misc/dbt-materialize/dbt/adapters/materialize/connections.py [hunk 1]
```diff
@@ -47,6 +47,18 @@ def connect(**kwargs):
         # dbt prints notices to stdout, which is very distracting because dbt
         # can establish many new connections during `dbt run`.
         "--welcome_message=off",
+        # Disable warnings about the session's default database or cluster not
+        # existing, as these get quite spammy, especially with multiple threads.
+        #
+        # Details: it's common for the cluster specified in profiles.yml (often
+        # `quickstart`) to be absent. For many dbt deployments, clusters are
```
Comment (morsapaes) [inline misc/dbt-materialize/dbt/adapters/materialize/connections.py:54]: Does this matter? We don't use the cluster specified in `profiles.yml` to create the connection, which is why these notices end up being really annoying. IIRC, passing the target cluster in connections wasn't supported in `libpq`, last we checked!

**Review by morsapaes — APPROVED** [review]

**Review by jkosh44 — APPROVED** [review]
Personally, I'd prefer implementing something like `client_min_messages` that allows clients to turn off all notices. However, I think this is fine for now.

File: misc/dbt-materialize/dbt/adapters/materialize/connections.py (hunk 1 above)
Comment (benesch) [reply misc/dbt-materialize/dbt/adapters/materialize/connections.py]: Ah, good point. Updated the comment accordingly.

> passing the target cluster in connections wasn't supported in `libpq`, last we checked!

It is supported in libpq and psycopg2, via `options="--cluster=<NAME>"`—it's the `PostgresConnectionManager` that doesn't make it easy to override just `options` —you have to rip and replace the whole thing, which is a good bit of work.

**Comment by benesch** [comment]
> Personally, I'd prefer implementing something like `client_min_messages` that allows clients to turn off all notices. However, I think this is fine for now.

I did consider this! However, I'm not sure whether the dbt adapter wants to turn off _all_ notices. Some of the remaining notices are plausibly valuable.

**Pull request merged**